src/App.tsx

```diff
@@ -185,12 +192,50 @@ class App extends Component {
     this.setState({shareId});
     database.setShareId(shareId);
 
-    const updatedNewContext = await getDataContext(dataContextName);
+    const updatedNewContext = await Codap.getDataContext(dataContextName);
     if (updatedNewContext) {
       database.set("dataContext", updatedNewContext);
     }
   }
 
+  joinShare = async () => {
+    const shareId = this.state.joinShareId;
+    this.setState({shareId});
+    database.setShareId(shareId);
+
+    const response = await database.getAll();
+    const contextData = response && response.val();
+    if (contextData) {
+      const { dataContext, items } = contextData;
+
+      // fix parent references -- assumes collections are written out in order
+      const collectionNames: string[] = [];
+      dataContext.collections.forEach((collection: any, index: number) => {
+        collectionNames.push(collection.name);
+        if (index > 0) {
+          collection.parent = collectionNames[index - 1];
+        }
+      });
+      await Codap.createDataContext(dataContext);
+
+      // combine items from all users in a single array
+      const allItems = [];
+      // tslint:disable-next-line: forin
+      for (const label in items) {
```

Comment:
We could use this opportunity to push the user's own label to the bottom.

Reply:
Good point